src/semantic_world/geometry.py

```diff
@@ -76,12 +85,53 @@ class Shape(ABC):
     """
     origin: TransformationMatrix = field(default_factory=TransformationMatrix)
 
-    def as_bounding_box(self) -> BoundingBox:
+    def as_bounding_box(self, reference_frame: Optional[Body]) -> BoundingBox:
```

Comment:
it's feels weird that reference frame can be optional. you are computing it using information from this shape, so it should always have the same reference frame as the shape, no?
btw, even if this stays optional, it should have default None, otherwise its not really optional

Reply:
can you give me an example?

Reply:
hm idk, doesn't that violate the single responsibility principle?